uit/gui_tools/submit.py

```diff
@@ -75,8 +78,24 @@ def env_var_widget(self, val, tag, **kwargs):
         widget.param.watch(self.update_environ, ['value'], onlychanged=True)
         return widget
 
-    def toggle_file_browser(self):
-        self.browse_toggle = not self.browse_toggle
+    @param.depends()
+    def env_file_browser_widget(self, tag, **kwargs):
+        widget = pn.widgets.Button(name='📂', css_classes=[tag], width=30, align='end', **kwargs)
+        widget.on_click(self.toggle_file_browser)
+        return widget
+
+    def toggle_file_browser(self, *events):
+        for event in events:
+            _, is_key, i = event.obj.css_classes[0].split('_')
+            i = int(i)
+            new_val = self.env_values[i]
+            new_browser_pn = HpcFileBrowser(self.uit_client)
+            new_browser_pn.callback = partial(self.new_callback, a=new_val, b=new_browser_pn)
+            self.browse_view[:] = [new_browser_pn.panel]
+
+    @staticmethod
+    def new_callback(new_selection, a, b):
+        a.value = b.value[0]
 
     @param.depends('browse_toggle')
     def file_browser_view(self):
```

Comment:
This method is no longer needed and can be removed.

Reply:
Removed, thank you.